Products/ZenUI3/browser/resources/js/zenoss/support.js

```diff
@@ -0,0 +1,189 @@
+                                        win.show();
+                                    }
+                                });
+                            }
+                        });
+                        dialog.show();
+                    }
+                },{
+                    itemId: 'deletebundles-button',
+                    iconCls: 'delete',
+                    text: _t('Delete'),
+                    tooltip: _t('Delete Support Bundles'),
+                    disabled: true,
+                    handler: function() {
+                        var grid = Ext.ComponentQuery.query('#supportbundles')[0];
+                        bundleNames = [];
+                        Ext.each(grid.getSelectionModel().getSelection(), function(row) {
+                            bundleNames.push(row.data.fileName);
+                        });
+
+                        Zenoss.remote.SupportRouter.deleteBundles({bundleNames:bundleNames}, function() {
+                            grid.refresh();
+                        });
+
+                    }
+                },{
+                    itemId: 'downloadbundles-button',
+                    iconCls: 'export',
+                    text: _t('Download'),
+                    tooltip: _t('Download support bundle(s)'),
+                    disabled: true,
+                    handler: function() {
+                        var grid = Ext.ComponentQuery.query('#supportbundles')[0];
+                        Ext.each(grid.getSelectionModel().getSelection(), function(row) {
+                            console.log(row.data.fileName);
+                            var form = Ext.create('Ext.form.Panel', {
+                                renderTo: Ext.getBody(),
+                                standardSubmit: true,
+                                url: '/zport/dmd/getSupportBundle?bundle=' + row.data.fileName
+                            });
+                            form.submit({
+                                target: '_blank',
+                            });
+                            Ext.defer(function(){
+                                form.close();
+                            }, 100);
+                        });
+                    }
+                },'->',{
+                    itemId: 'refreshbundles-button',
+                    xtype: 'refreshmenu',
+                    ref: 'refreshmenu',
+                    stateId: 'bundlesrefresh',
+                    iconCls: 'refresh',
+                    text: _t('Refresh'),
+                    tooltip: _t('Refresh Support Bundles List'),
+                    handler: function() {
+                        var grid = Ext.ComponentQuery.query('#supportbundles')[0];
+                        grid.refresh();
+                    }
+                }]
+            },
+            listeners: {
+                selectionchange: function() {
+                    var grid = Ext.ComponentQuery.query('#supportbundles')[0];
```

Comment:
maybe cache the "supportbundles" element on this component so that you dont have to query the DOM for it each time you wanna do something with it.